### PR TITLE
Allow the url option to be dynamically generated (Fixed)

### PR DESCRIPTION
**Pull request by tm1000**
#88 was a pretty trivial change, since the user never returned I decided to take what @boneyao said in #88 and make the corrections. I am submitting this request to replace #88

### CODE REVIEW

**Comment by truckingsim** [comment]
@tm1000 there's an error thrown when trying to use this:

```js
ajax-bootstrap-select.js:954 Uncaught TypeError: Cannot read property 'url' of undefined
    at new AjaxBootstrapSelectRequest (ajax-bootstrap-select.js:954)
    at ajax-bootstrap-select.js:372
AjaxBootstrapSelectRequest @ ajax-bootstrap-select.js:954
(anonymous) @ ajax-bootstrap-select.js:372
setTimeout (async)
(anonymous) @ ajax-bootstrap-select.js:365
dispatch @ jquery.min.js:3
v.handle @ jquery.min.js:3
```

We need to check for `this.options.ajax` as well.  Can you fix that please.

**Comment by tm1000** [comment]
Ack! What a trivial error!

**Comment by tm1000** [comment]
Fixed. However it's three commits now. So let's get it working and then I will squash them all into one.

**Comment by truckingsim** [comment]
@tm1000 we don't need to check typeof if you are just checking for undefined.  `undefined` is a falsey value.  You can just do:

```js
if(this.options.ajax && this.options.ajax.url && ...etc)
```

**Comment by tm1000** [comment]
Ah. Never realized that. Well then.. update coming

**Comment by tm1000** [comment]
Done. Thanks for you guidance.

**Comment by truckingsim** [comment]
@tm1000 this still isn't working for me 

```js
http://localhost/Ajax-Bootstrap-Select/example/php/function%20()%20%7B%20return%20'ajax.php';%20%7D 
```

Its stringifying the function code and putting that whole stringified code as the url.  Don't have time right now to look at this, will be later today or tomorrow if you can't figure it out.  It would be great if you could give it a try though.

**Comment by tm1000** [comment]
Hi @truckingsim 

I'll try to look a bit later when I implement your library and have a play. Thanks

File: src/js/classes/AjaxBootstrapSelectRequest.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ var AjaxBootstrapSelectRequest = function (plugin) {
             q: '{{{q}}}'
         };
     }
+    
+    if (this.options.ajax && this.options.ajax.url && $.isFunction(this.options.ajax.url)) {
```
Comment (markcarver) [inline src/js/classes/AjaxBootstrapSelectRequest.js]: This is wrong. It should be checking if `this.options.url` exists and is a function.

The `this.options` in this context is for the `AjaxBootstrapSelectRequest` class, not the plugin's options:

See https://github.com/truckingsim/Ajax-Bootstrap-Select/pull/102/files#diff-192c7a5fedc370a66ab88b12003d5f23L30

```js
this.options = $.extend(true, {}, plugin.options.ajax);
```

**Pull request closed without merging**